internal/services/mssql/mssql_database_resource.go

```diff
@@ -251,125 +260,132 @@ func resourceMsSqlDatabaseCreate(d *pluginsdk.ResourceData, meta interface{}) er
+	for _, s := range databases.PossibleValuesForDatabaseStatus() {
+		if s != string(databases.DatabaseStatusOnline) {
+			pendingStatuses = append(pendingStatuses, s)
 		}
 	}
+
+	deadline, ok := ctx.Deadline()
+	if !ok {
+		return fmt.Errorf("internal-error: context had no deadline")
+	}
+
 	stateConf := &pluginsdk.StateChangeConf{
 		Pending: pendingStatuses,
-		Target:  []string{string(sql.DatabaseStatusOnline)},
+		Target:  []string{string(databases.DatabaseStatusOnline)},
 		Refresh: func() (interface{}, string, error) {
 			log.Printf("[DEBUG] Checking to see if %s is online...", id)
 
-			resp, err := client.Get(ctx, id.ResourceGroup, id.ServerName, id.Name)
+			resp, err := client.Get(ctx, id, databases.DefaultGetOperationOptions())
 			if err != nil {
 				return nil, "", fmt.Errorf("polling for the status of %s: %+v", id, err)
 			}
 
-			if props := resp.DatabaseProperties; props != nil {
-				return resp, string(props.Status), nil
+			if resp.Model != nil && resp.Model.Properties != nil && resp.Model.Properties.Status != nil {
+				return resp, string(pointer.From(resp.Model.Properties.Status)), nil
 			}
 
 			return resp, "", nil
 		},
-		MinTimeout:                1 * time.Minute,
 		ContinuousTargetOccurence: 2,
+		MinTimeout:                1 * time.Minute,
+		Timeout:                   time.Until(deadline),
 	}
 
-	stateConf.Timeout = d.Timeout(pluginsdk.TimeoutCreate)
-
 	if _, err = stateConf.WaitForStateContext(ctx); err != nil {
 		return fmt.Errorf("waiting for %s to become ready: %+v", id, err)
 	}
 
 	// Cannot set transparent data encryption for secondary databases
-	if createMode != string(sql.CreateModeOnlineSecondary) && createMode != string(sql.CreateModeSecondary) {
-		statusProperty := sql.TransparentDataEncryptionStatusDisabled
+	if createMode != string(databases.CreateModeOnlineSecondary) && createMode != string(databases.CreateModeSecondary) {
+		stateProperty := transparentdataencryptions.TransparentDataEncryptionStateDisabled
 		encryptionStatus := d.Get("transparent_data_encryption_enabled").(bool)
 		if encryptionStatus {
-			statusProperty = sql.TransparentDataEncryptionStatusEnabled
+			stateProperty = transparentdataencryptions.TransparentDataEncryptionStateEnabled
 		}
-		_, err := transparentEncryptionClient.CreateOrUpdate(ctx, id.ResourceGroup, id.ServerName, id.Name, sql.TransparentDataEncryption{
-			TransparentDataEncryptionProperties: &sql.TransparentDataEncryptionProperties{
-				Status: statusProperty,
+		_, err := transparentEncryptionClient.CreateOrUpdate(ctx, id, transparentdataencryptions.LogicalDatabaseTransparentDataEncryption{
+			Properties: &transparentdataencryptions.TransparentDataEncryptionProperties{
+				State: stateProperty,
 			},
 		})
 		if err != nil {
 			return fmt.Errorf("while enabling Transparent Data Encryption for %q: %+v", id.String(), err)
 		}
 
 		if err = pluginsdk.Retry(d.Timeout(pluginsdk.TimeoutCreate), func() *pluginsdk.RetryError {
```

Comment:
another internal x-ref, this is another case of https://github.com/hashicorp/go-azure-sdk/issues/307 so this can be removed once that's fixed

Reply:
Note added, fixed.